src/services/post.ts

```diff
@@ -1,43 +1,52 @@
+//Resource: https://stackoverflow.com/questions/48036975/firestore-multiple-conditional-where-clauses
+
 import * as firebase from "firebase/app";
 import 'firebase/storage';
 import { Post } from '../shared/models/post.model';
-import { authenticateFromStore } from "./user";
+import {authenticateFromStore, getUserDataByID, getUserRoleByID} from "./user";
+import {roles} from "../enums/enums";
+import {Simulate} from "react-dom/test-utils";
 
 export const getPosts = async (): Promise<Post[]> => {
   await authenticateFromStore();
   const user = firebase.auth().currentUser;
   const db = firebase.firestore();
-
   const posts: Array<Post> = [];
 
-  await db.collection("posts")
-                        .where("creatorID", "==", user?.uid).orderBy("date", "desc").get()
```

Comment:
I'd recommend adding the orderBy bit back in

Reply:
Good suggestion - done!